### PR TITLE
Support for mirror option in sync tasks

### PR DESCRIPTION
**Pull request by vonsch**

### CODE REVIEW

**Comment by vonsch** [comment]
Please note I tested only pulp_rpm codepath in our environment, but I'm 99% sure that the other code paths will work as well

File: plugins/module_utils/pulp.py [hunk 1]
```diff
@@ -213,11 +213,15 @@ def delete(self):
...
         self.module.set_changed()
 
-    def sync(self, remote_href):
+    def sync(self, remote_href, mirror):
         if not hasattr(self, "_sync_id"):
             raise SqueezerException("This entity is not syncable.")
+
+        body = {"remote": remote_href}
+        if mirror:
+            body["mirror"] = mirror
```
Comment (mdellweg) [inline plugins/module_utils/pulp.py:222]:
```suggestion
        if mirror is not None:
            body["mirror"] = mirror
```
If we also send mirror=False, we can overwrite (on demand) whatever the server sets as default.

File: plugins/modules/ansible_sync.py [hunk 2]
```diff
@@ -35,6 +35,10 @@
       - Name of the repository
     type: str
     required: true
+  mirror:
+    description:
+      - If True, synchronization will run in "mirror" mode instead of "additive"
```
Comment (mdellweg) [inline plugins/modules/ansible_sync.py:40]: Maybe explain that mirror mode removes content that disappeared upstream.
Also False means additive (no content is removed).
If not specified the server default is used.
(see my other comment for this tri-state)

**Review by mdellweg — COMMENTED** [review]
Thank you for this addition.
Would you mind to add a test for this, too?

**Comment by vonsch** [comment]
Obsoleted by https://github.com/pulp/squeezer/pull/86

**Pull request closed without merging**